### DISCUSSION TITLE
Unattended install of WSL2

### DISCUSSION THREAD

**Discussion by kilrau** [microsoft/WSL#6201] [General] [answered]
Continuation of https://github.com/microsoft/WSL/issues/3369:
I'd appreciate being pointed at guides, scripts and anything that helps to figure out how to get a fully unattended install of WSL2 working on Windows 2004 or higher.

**Comment by ghost** [comment, marked as answer]
I think the "coming soon to stable releases" "wsl --install" would solve this problem

**Reply by kilrau** [reply]
Yes, that looks like the solution indeed! Thank you!

According to https://devblogs.microsoft.com/commandline/distro-installation-added-to-wsl-install-in-windows-10-insiders-preview-build-20246/ the feature is already available in the dev channel.

How can I find out in which Windows version this will be available once on stable? In particular I'd be interested if versions down to 1903 will be getting this feature. @craigloewen-msft

**Reply by ghost** [reply]
If this answer helped you, you could mark this comment as the answer

**Comment by kilrau** [comment]
Is there any way to achieve something similar with the current wsl.exe installer?

**Reply by craigloewen-msft** [reply]
Yes, Ubuntu is available in the winget-pkgs repo.

**Reply by ghost** [reply]
@craigloewen-msft I was talking about the WSL Linux Kernel package

**Reply by kilrau** [reply]
Formulated a bit more abstract, the questions are:
1. Is there another way to install the WSL2 Linux Kernel package **unattended** with current Windows stable which do *not* have the `wsl --install` feature yet?
2. What is the ETA for `wsl --install` to land on the stable channel?

**Reply by craigloewen-msft** [reply]
I don't believe the Linux kernel package is part of the winget-pkgs repo, it needs to be downloaded from the link we've provided in the docs.
To answer the questions above:

1. You could download the MSI directly using PowerShell and install it an Administrator window. 
2. There isn't a set ETA yet for `wsl --install` to land on the stable channel.
